cpp/src/stream_compaction/distinct_reduce.hpp

```diff
@@ -68,11 +66,13 @@ auto constexpr reduction_init_value(duplicate_keep_option keep)
  * @param has_nested_columns Indicates whether the input table has any nested columns
  * @param keep The parameter to determine what type of reduction to perform
  * @param nulls_equal Flag to specify whether null elements should be considered as equal
+ * @param nans_equal Flag to specify whether NaN values in floating point column should be
+ *        considered equal.
  * @param stream CUDA stream used for device memory operations and kernel launches
  * @param mr Device memory resource used to allocate the returned vector
  * @return A device_uvector containing the reduction results
  */
-rmm::device_uvector<size_type> hash_reduce_by_row(
+rmm::device_uvector<size_type> distinct_reduce(
```

Comment:
`distinct` already implies removing duplicates within the group (or reduction) thus `distinct_reduce` doesn't sound right to me.

Reply:
Do you have a suggestion for the new name?

Reply:
Probably I'll merge this into `distinct.cu` if compiling this file is not an issue.

Reply:
Alright, merging this inflates the compile time from 40s to 90s. Not too bad if compared with many minutes of compilation time of many others. Then I'll merge it.

Reply:
Ah good point. Build time is another reason that we are doing this file splitting. Maybe `reduce_by_row` or `reduce_helper`?